Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/DomainDaoImpl.java

```diff
@@ -310,25 +313,18 @@ public Domain getDomainByName(String name) {
     @Override
     @SuppressWarnings("unchecked")
     public List<Domain> listDomains(String name, int count) {
-        // Direct RI Comment:
-        // TODO I'm not sure if this is doing the right thing. I suspect that the
-        // real intent is to do some kind of db paging
-
+        // Direct RI Comment: Use by Direct third party component
+        //(org.nhindirect.gateway.smtp.config.WSSmtpAgentConfig), it will pass null,maxcount
         List<Domain> results = null;
         Query query;
 
         Session session = null;
 
         try {
             session = DaoUtils.getSession();
-
+            log.debug("Calling listDomain");
             query = session.getNamedQuery("getDomainsByName");
-            String nameUpperCase = "";
-            if (name != null) {
-                nameUpperCase = name.toUpperCase(Locale.getDefault());
-            }
-
-            query.setParameter("domainName", nameUpperCase);
+            query.setParameter("domainName", StringUtils.hasText(name) ? name.toUpperCase(Locale.getDefault()): null);
```

Comment:
Can you add a comment right before this line saying that the parameter must be null if name is null.  Something like "In case no name exists, the parameter must be null and not an empty String to allow AutomatedDirectTest SoapUI test to finish."